subsys/bluetooth/shell/bt.c

```diff
@@ -434,12 +434,30 @@ static void per_adv_sync_sync_cb(struct bt_le_per_adv_sync *sync,
 				 struct bt_le_per_adv_sync_synced_info *info)
 {
 	char le_addr[BT_ADDR_LE_STR_LEN];
+	char past_peer[BT_ADDR_LE_STR_LEN];
 
 	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	if (info->conn) {
+		conn_addr_str(info->conn, past_peer, sizeof(past_peer));
+	} else {
+		memset(past_peer, 0, sizeof(past_peer));
+	}
+
 	shell_print(ctx_shell, "PER_ADV_SYNC[%u]: [DEVICE]: %s synced, "
-		    "Interval 0x%04x (%u ms), PHY %s",
+		    "Interval 0x%04x (%u ms), PHY %s, SD 0x%04X, PAST peer %s",
 		    bt_le_per_adv_sync_get_index(sync), le_addr,
-		    info->interval, info->interval * 5 / 4, phy2str(info->phy));
+		    info->interval, info->interval * 5 / 4, phy2str(info->phy),
+		    info->service_data, past_peer);
+
+	if (info->conn) { /* if from PAST */
+		for (int i = 0; i < ARRAY_SIZE(per_adv_syncs); i++) {
+			if (!per_adv_syncs[i]) {
+				per_adv_syncs[i] = sync;
+				break;
+			}
+		}
```

Comment:
What happens if the loop terminates without a match?   Is that possible?

Reply:
In theory: No :) 

If the shell indeed does not fit the sync pointer, then something went wrong in the host stack, as missing "free" slots should have stopped the PAST from ever reaching the (shell) application.